src/ssl.c

```diff
@@ -17813,24 +17837,131 @@ size_t wolfSSL_get_client_random(const WOLFSSL* ssl, unsigned char* out,
 #endif
     }
 
+    /* If a valid struct is provided with function pointers, will override
+       RAND_seed, bytes, cleanup, add, pseudo_bytes and status.  If a NULL
+       pointer is passed in, it will cancel any previous function overrides.
+
+       Returns WOLFSSL_SUCCESS on success, WOLFSSL_FAILURE on failure. */
+    int wolfSSL_RAND_set_rand_method(const WOLFSSL_RAND_METHOD *methods)
+    {
+        int ret = WOLFSSL_FAILURE;
+
+        if (initRandMethMutex == 0) {
+            if (wc_InitMutex(&rand_methods_mutex) != 0) {
+                WOLFSSL_MSG("Bad Init Mutex rand methods");
+                ret = BAD_MUTEX_E;
+            }
+            else {
+                initRandMethMutex = 1;
+            }
+        }
+
+        if (initRandMethMutex != 0) {
+            if (wc_LockMutex(&rand_methods_mutex) != 0) {
+                WOLFSSL_MSG("Lock rand methods mutex failed");
+                ret = BAD_MUTEX_E;
+            }
+            else {
+                rand_methods = methods;
+                wc_UnLockMutex(&rand_methods_mutex);
+                ret = WOLFSSL_SUCCESS;
+            }
+        }
+
+        return ret;
+    }
+
+    /* Sets func_ptr to a pointer to the given function func if it was set
+       by RAND_set_rand_method, or NULL if either it was not set,
+       or there was an error retreiving it.
+
+       Returns WOLFSSL_SUCCESS on success, otherwise returns an error code. */
+    static int GetRandFunction(RandFunction func, RandVoidMethod** func_ptr)
```

Comment:
This was not what I had in mind. This code is not smaller. Now there is a new `enum RandFunction` and function to get a callback with a void*. I only wanted the mutex init code in a static. Sending you a direct note and will provide you some reference code for what I had in mind.